src/daft-sql/src/modules/utf8.rs

```diff
@@ -539,3 +541,28 @@ impl SQLFunction for SQLTokenizeDecode {
         }
     }
 }
+
+pub struct SQLConcat;
+
+impl SQLFunction for SQLConcat {
+    fn to_expr(
+        &self,
+        inputs: &[sqlparser::ast::FunctionArg],
+        planner: &crate::planner::SQLPlanner,
+    ) -> SQLPlannerResult<ExprRef> {
+        let inputs = inputs
+            .iter()
+            .map(|input| planner.plan_function_arg(input))
+            .collect::<SQLPlannerResult<Vec<_>>>()?;
+        let mut inputs = inputs.into_iter();
+
+        let Some(mut first) = inputs.next() else {
+            invalid_operation_err!("concat requires at least one argument")
+        };
+        for input in inputs {
+            first = binary_op(Operator::Plus, first, input);
+        }
+
+        Ok(first)
+    }
+}
```

Comment:
There's a lot of different things that are called concat in Daft (the agg expr, concatting two tables, and now this). Would appreciate some doc strings just to make it easier to understand at a glance what this one does, and why it uses the plus operator.